Objects/tupleobject.c

```diff
@@ -186,6 +210,15 @@ PyTuple_Pack(Py_ssize_t n, ...)
 static void
 tupledealloc(PyTupleObject *op)
 {
+    if (_PyTuple_HAS_NO_GC(op)) {
+        Py_ssize_t i = PyTuple_GET_SIZE(op);
+        while (--i >= 0) {
+            Py_XDECREF(op->ob_item[i]);
+        }
+        Py_TYPE(op)->tp_free((PyObject *)op);
+        return;
+    }
```

Comment:
We should fail if size is 0.  So this should go after the `if (Py_SIZE(op) == 0) {` block.